New package: Orbmu2k.nvidiaProfileInspector version 2.4.0.16

ToasterDEV opened the pull request:
Checklist for Pull Requests
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [ ] Is there a linked Issue?

Manifests
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/doc/Authoring.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.9 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.9.0)?

Note: `<path>` is the directory's name containing the manifest you're submitting.

---

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/230186)

ToasterDEV committed:
New package: Orbmu2k.nvidiaProfileInspector version 2.4.0.16

stephengillie reviewed (changes requested):
Hi @ToasterDEV,

This package appears to have a dependency:

![image](https://github.com/user-attachments/assets/e5791f38-1c36-46c6-8785-cfaac8643bf5)

ToasterDEV committed:
Update manifests/o/Orbmu2k/nvidiaProfileInspector/2.4.0.16/Orbmu2k.nvidiaProfileInspector.installer.yaml

Missed adding a dependency to the manifest (.NET Framework 4.8.1), thankfully it was noticed before merging.

Co-authored-by: Stephen Gillie <114952565+stephengillie@users.noreply.github.com>

ToasterDEV commented:
> Hi @ToasterDEV,
> 
> This package appears to have a dependency:
> 
> ![image](https://private-user-images.githubusercontent.com/114952565/413624185-e5791f38-1c36-46c6-8785-cfaac8643bf5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzk3MzkxNDEsIm5iZiI6MTczOTczODg0MSwicGF0aCI6Ii8xMTQ5NTI1NjUvNDEzNjI0MTg1LWU1NzkxZjM4LTFjMzYtNDZjNi04Nzg1LWNmYWFjODY0M2JmNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIxNlQyMDQ3MjFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lMGVmNzVmZThiMzQ5NTdkMjczOWYwM2I1NWM1YTZkMjdjNmI5MDgwNTJjYjBiNjZmODRjYmJlZTA4YjczM2JjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.vp6htB-4L0jajOcIggDbyTe9FLF3iUdroMDNAQxiOGc)

Hmm, while I can see that installation on my end does succeed when using `Microsoft.DotNet.Framework.DeveloperPack_4` as a dependency (see code block below)

```
Found NVIDIA Profile Inspector [Orbmu2k.nvidiaProfileInspector] Version 2.4.0.16
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
This package requires the following dependencies:
  - Packages
      Microsoft.DotNet.Framework.DeveloperPack_4
Successfully verified installer hash
Extracting archive...
Successfully extracted archive
Starting package install...
Command line alias added: "nvidiaProfileInspector"
Successfully installed
```
The required version of the .NET Framework shown in the screenshot seems to be a Windows Component that usually comes enabled by default ([NetFx4](https://learn.microsoft.com/en-us/windows/compatibility/net-framework-4-5-is-default)).
```
Get-WindowsOptionalFeature -Online -FeatureName NetFx4

FeatureName      : NetFx4
DisplayName      : .NET Framework 4.8
Description      : .NET Framework 4.8
RestartRequired  : Possible
State            : Enabled
CustomProperties :
                   ServerComponent\Description : .NET Framework 4.8 provides a comprehensive and consistent programming model for quickly and easily building and running applications that are built for various platforms including desktop PCs, Servers, smart phones and the public and private cloud.
                   ServerComponent\DisplayName : .NET Framework 4.8
                   ServerComponent\Id : 418
                   ServerComponent\Type : Feature
                   ServerComponent\UniqueName : NET-Framework-45-Core
                   ServerComponent\Deploys\Update\Name : NetFx4
                   ServerComponent\Version\Major : 4
                   ServerComponent\Version\Minor : 5
```

Trying to declare this component as a dependency seems to do the trick, though at least on my end it results in a CBS Exception (CBS_E_INVALID_INSTALL_STATE) from what I'm able to gather given I already have it enabled.

```
Found NVIDIA Profile Inspector [Orbmu2k.nvidiaProfileInspector] Version 2.4.0.16
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
This package requires the following dependencies:
  - Windows Features
      NetFx4
Enabling [NetFx4]...
Failed to enable [NetFx4] feature: 2148468755
Failed to enable Windows Feature dependencies. To proceed with installation, use '--force'.
```
Considering that the feature name is parsed as valid and that the rest of the manifest appears to work as intended, should we change the dependency to the corresponding Windows Feature, instead of requiring the installation of the Developer Pack?

```
Dependencies:
  WindowsFeatures:
    - NetFx4
```

stephengillie commented:
I was unaware of `NetFx4`. Please feel free to update as you see fit.

Trenly reviewed (commented):
Does this application require an NVIDIA graphics card?

ToasterDEV committed:
Update Orbmu2k.nvidiaProfileInspector.installer.yaml to require NetFx4

Modified the installation manifest to call upon a Windows Optional Feature that corresponds to .NET Framework 4.8.1 (NetFx4).

This feature should come enabled by default since Windows 8.1, but the application will not run without it, so we'll check for it just in case.

ToasterDEV commented:
> I was unaware of `NetFx4`. Please feel free to update as you see fit.

Got it! Just sent an updated commit to handle the requirement.

> Does this application require an NVIDIA graphics card?

Yes, it's designed to aid in getting functionality within the driver stack to apply to applications that don't explicitly request specific features, or modify the ones that do.

Trenly commented:
> Yes, it's designed to aid in getting functionality within the driver stack to apply to applications that don't explicitly request specific features, or modify the ones that do.

[Policy] Hardware

ToasterDEV commented:
> > Yes, it's designed to aid in getting functionality within the driver stack to apply to applications that don't explicitly request specific features, or modify the ones that do.
> 
> [Policy] Hardware

Okay, wait a moment please.

While the application is designed to aid in interacting with the NVIDIA driver stack, it does not require the presence of any device to be able to run and/or perform its designated function.

Off the top of my head, I can think of at least two cases in which the application might be desired without the presence of any graphics card installed (those being devices using external PCIe enclosures to connect external graphics cards, and also computers that might use more than one vendor to perform rendering at once).

Besides the previous two cases, there are several applications currently available in the WinGet repository that are able to go through the installation process without requiring the physical hardware being present at the moment of the installation, such as the following:

- 9NF8H0H7WMLT (NVIDIA Control Panel)
- Nvidia.CUDA
- Nvidia.FrameView
- Nvidia.Omniverse
- Nvidia.PhysX
- Nvidia.PhysXLegacy
- Nvidia.RTXVoice

This application in particular can work without the physical hardware being present, as it is just facilitating the modification of a database storing properties and optimizations for other software that **does** require the hardware to be physically present at runtime, but it as of _itself_ doesn't depend on any particular piece of hardware.

Trenly commented:
Reopen with reason: Automation error;

ToasterDEV commented:
> Manual Validation ended with:
> 
> > 8:17:21 AM: 2025-02-18 08:17:06.865 [CORE] Failed to enable Windows Feature [NetFx4] with exit code: 2148468755
> 
> (Automated response - build 982.)

Hmm, odd.

Just to be sure, I set up a new Hyper-V VM with Windows 11 as the guest, and after getting to the desktop, I tried to run WinGet v1.6.10121 with the following results:
![image](https://github.com/user-attachments/assets/f46e81be-23e4-44a3-975e-04e4f6d9e9d4)

However, if I try to repeat the procedure after updating to the latest WinGet version (v1.9.25200), the aforementioned issue becomes reproducible:

![image](https://github.com/user-attachments/assets/cb48ffd1-d3eb-4140-9ae1-0c02da73a69e)

I believe this might be an issue with winget-cli itself, though I'm not sure how to best narrow it down when creating an issue in the correct repository. Should I just report what we saw here as-is?

ItzLevvie commented:
Close with reason: Stale pull request;
